iOS/Flows/People/PeopleCoordinator.swift

```diff
@@ -36,9 +43,230 @@ class PeopleCoordinator: PresentableCoordinator<[Person]> {
             self.peopleNavController.prepareForInvitations()
             
             Task {
-                await self.updateInvitation()
+                await self.inviteRemainingPeople()
             }.add(to: self.taskPool)
         }
     }
 }
 
+extension PeopleCoordinator {
+
+    /// Invites the next person in the array of people to invite and keeps doing so until there are no more people to invite.
+    /// Once there are no more people to invite, the finishInviting phase is called.
+    @MainActor
+    func inviteRemainingPeople() async {
+        // If there are no more people to invite, then finish the flow.
+        guard var person = self.peopleToInvite[safe: self.inviteIndex] else {
+            await self.finishInviting()
+            return
+        }
+
+        guard !Task.isCancelled else { return }
+
+        self.inviteIndex += 1
+
+        // If the invited person is already connected, there's no need to formally invite them.
+        // Move on to the next person
+        if let connection = try? await person.connection?.retrieveDataIfNeeded(),
+            connection.status == .accepted {
+            // Do nothing because the person is already connected.
+        } else {
+            // The person is not connected, we'll need to try inviting them to connect.
+            if let connection = await self.invite(person: person) {
+                // If a connection was made, update the person's connection info.
+                person.connection = connection
+            }
+        }
+
+        // Keep track of who was connected/invited.
+        self.invitedPeople.append(person)
+
+        // Continue inviting the rest of the people.
+        await self.inviteRemainingPeople()
+    }
+
+    func invite(person: Person) async -> Connection? {
+        await self.peopleNavController.peopleVC.showLoading(for: person)
+
+        // This person is not in you contacts, so you can't invite them.
+        guard let contact = person.cnContact else { return nil }
+
+        // If the user already has an account, we can just connect with them directly.
+        if let user = await self.findUser(with: contact) {
+            return await self.presentConnectionFlow(for: user)
+        } else if let reservation = self.getReservation(for: contact) {
+            // Get a valid reservation object that we can use to invite this person. Then prompt the user
+            // to send them a text
+            await self.sendText(to: contact, with: reservation)
+        }
+
+        return nil
+    }
+
+    /// Returns the reservation that should be used to invite this person. Nil is returned if there are no valid reservation objects left that can be used.
+    private func getReservation(for contact: CNContact) -> Reservation? {
```

Comment:
Previously we weren't checking to see if a reservation was already assigned to someone. This fixes that.